logstash-core/lib/logstash/pipeline.rb

```diff
@@ -107,16 +107,23 @@ def compile_lir
     LogStash::Compiler.compile_sources(sources_with_metadata, @settings)
   end
 
-  def plugin(plugin_type, name, *args)
+  def plugin(plugin_type, name, line, column, *args)
     @plugin_counter += 1
 
     # Collapse the array of arguments into a single merged hash
     args = args.reduce({}, &:merge)
 
-    id = if args["id"].nil? || args["id"].empty?
-      args["id"] = "#{@config_hash}-#{@plugin_counter}"
-    else
-      args["id"]
+    # Pull the ID from LIR to keep IDs consistent between the two representations
+    id = lir.graph.vertices.filter do |v| 
```

Comment:
@andrewvc I admit this is a bit of a nitpick, but here it goes:

 Can't we generate that `id` in `config_ast` already and just pass it in here. `Pipeline` is already super huge and convoluted. It does seem very wrong to now also add logic to generate a plugin `id` from its position in the config string in here. Seems more reasonable to do this in `config_ast` which already deals with the config source itself doesn't it?